architectures/firmware/sof-zephyr/app_layer/index.rst

```diff
@@ -3,35 +3,40 @@
 Application Layer
 #################
 
-Application Layer represents the built-in FW processing components, loadable FW
-components and example templates with application libraries required for
-components integration. Application layer content is assumed to be open source
+Application Layer represents the built-in FW modules (processing components), loadable FW
+modules and example templates with application libraries required for
+modules integration. Application layer content is assumed to be open source
 and only proprietary 3rd party components should remain private.
 
 .. uml:: images/app_layer_diagram.pu
    :caption: Application Layer
 
-Components in Application Layer
+Modules in Application Layer
 *******************************
 
-The built-in components are built together with base firmware and they have
+The built-in modules are built together with base firmware and they have
 direct access to all firmware drivers and service APIs. When built-in module
 is enabled in configuration, it is guaranteed to exist in firmware binary.
 
-The loadable components are built separately from base firmware and they are
+The loadable modules are built separately from base firmware and they are
```

Comment:
separately but surely there's a need for common headers and some ABI version control?

Reply:
This is interesting topic. We tried to keep common headers but be faced cross dependencies with xtos what was troublesome.